src/transformers/models/layoutlmv2/modeling_layoutlmv2.py

```diff
@@ -1273,28 +1359,44 @@ def forward(
-
-        >>> image = Image.open("name_of_your_document - can be a png file, pdf, etc.").convert("RGB")
-        >>> question = "what's his name?"
-
-        >>> encoding = processor(image, question, return_tensors="pt")
-        >>> start_positions = torch.tensor([1])
-        >>> end_positions = torch.tensor([3])
-
-        >>> outputs = model(**encoding, start_positions=start_positions, end_positions=end_positions)
-        >>> loss = outputs.loss
-        >>> start_scores = outputs.start_logits
-        >>> end_scores = outputs.end_logits
-        ```"""
+        Example:
+
+        LayoutLMv2 depends on detectron2, torchvision and tesseract; passing doctests requires their installation. Run
+        the following to install them: `python -m pip install 'git+https://github.com/facebookresearch/detectron2.git'`
+        `python -m pip install torchvision tesseract`
+
+            ```python
+            >>> from transformers import LayoutLMv2Processor, LayoutLMv2ForQuestionAnswering, set_seed
+            >>> import torch
+            >>> from PIL import Image
+            >>> from datasets import load_dataset
+
+            >>> set_seed(88)
+            >>> processor = LayoutLMv2Processor.from_pretrained("microsoft/layoutlmv2-base-uncased")
+            >>> model = LayoutLMv2ForQuestionAnswering.from_pretrained("microsoft/layoutlmv2-base-uncased")
+
+            >>> dataset = load_dataset("hf-internal-testing/fixtures_docvqa")
+            >>> image_path = dataset["test"][0]["file"]
+            >>> image = Image.open(image_path).convert("RGB")
+            >>> question = "When is coffee break?"
+            >>> encoding = processor(image, question, return_tensors="pt")
+
+            >>> outputs = model(**encoding)
+            >>> predicted_start_idx = outputs.start_logits.argmax(-1).item()
+            >>> predicted_end_idx = outputs.end_logits.argmax(-1).item()
+            >>> predicted_start_idx, predicted_end_idx
+            (154, 287)
+
+            >>> predicted_answer_tokens = encoding.input_ids.squeeze()[predicted_start_idx : predicted_end_idx + 1]
+            >>> predicted_answer = processor.tokenizer.decode(predicted_answer_tokens)
+            >>> predicted_answer  # results are not very good without further fine-tuning
+            'council mem - bers conducted by trrf treasurer philip g. kuehn to get answers which the public refrigerated warehousing industry is looking for. plus questions from the floor. dr. emil m. mrak, university of cal - ifornia, chairman, trrf board ; sam r. cecil, university of georgia college of agriculture ; dr. stanley charm, tufts university school of medicine ; dr. robert h. cotton, itt continental baking company ; dr. owen fennema, university of wis - consin ; dr. robert e. hardenburg, usda. questions and answers exhibits open capt. jack stone'
+            ```
+
+            ```python
+            >>> target_start_index = torch.tensor([7]) >>> target_end_index = torch.tensor([14]) >>> outputs =
```

Comment:
This gives the following error when I ran the test

```
1484             ```python
1485             >>> target_start_index = torch.tensor([7]) >>> target_end_index = torch.tensor([14]) >>> outputs =
UNEXPECTED EXCEPTION: SyntaxError('invalid syntax', ('<doctest transformers.models.layoutlmv2.modeling_layoutlmv2.LayoutLMv2ForQuestionAnswering.forward[19]>', 1, 42, 'target_start_index = torch.tensor([7]) >>> target_end_index = torch.tensor([14]) >>> outputs =\n'))
Traceback (most recent call last):
  File "/home/yih_dar_huggingface_co/miniconda3/envs/py-3-9/lib/python3.9/doctest.py", line 1336, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest transformers.models.layoutlmv2.modeling_layoutlmv2.LayoutLMv2ForQuestionAnswering.forward[19]>", line 1
    target_start_index = torch.tensor([7]) >>> target_end_index = torch.tensor([14]) >>> outputs =
                                             ^
SyntaxError: invalid syntax
```

Reply:
Sorry about this -- the tear down function from `utils/prepare_for_doc_test.p` seems to have messed up the format